index.bs

```diff
@@ -1008,33 +1066,34 @@ parallel</a>:
 
   1. If the device is no longer connected to the system, <a>reject</a> |promise|
      with a {{NotFoundError}} and abort these steps.
-  2. Let |endpoint| be the IN endpoint in the <a>active configuration</a> with
+  1. Let |endpoint| be the IN endpoint in the <a>active configuration</a> with
      <code>bEndpointAddress</code> corresponding to |endpointNumber|. If there
      is no such endpoint <a>reject</a> |promise| with a {{NotFoundError}} and
      abort these steps.
-  3. If |endpoint| is not a bulk or interrupt endpoint <a>reject</a> |promise|
+  1. If |endpoint| is not a bulk or interrupt endpoint <a>reject</a> |promise|
      with an {{InvalidAccessError}} and abort these steps.
-  4. If <code>|device|.{{USBDevice/opened}}</code> or
+  1. If <code>|device|.{{USBDevice/opened}}</code> or
      <code>|interface|.{{USBInterface/claimed}}</code> is not <code>true</code>,
      <a>reject</a> |promise| with an {{InvalidStateError}} and abort these
      steps.
-  5. As appropriate for |endpoint| enqueue a bulk or interrupt IN transfer on
-     |endpoint| with a buffer sufficient to receive |length| bytes of data from
-     the device.
-  6. Let |result| be a new {{USBInTransferResult}}.
-  7. If data is returned as part of this transfer let |buffer| be a new
-     {{ArrayBuffer}} of exactly the length of the data received and set
-     <code>|result|.{{USBInTransferResult/data}}</code> to a new {{DataView}}
-     constructed over |buffer|.
-  8. If the device responds with more than |length| bytes of data set
+  1. Let |buffer| be a host buffer with space for |length| bytes.
+  1. As appropriate for |endpoint| enqueue a bulk or interrupt IN transfer on
```

Comment:
Sort of. The concept is spread across the USB protocol spec which defines the state machine for a transfer and the USB host controller specs which define how to instruct the host hardware to drive that state machine. Each OS is free to expose whatever interface it wants for programming the host controller but all of them end up looking the same.

The best way to clarify this seems to be to add a section specifying the operations that this specification expects the host system to provide and how those map to USB specification concepts (sending and receiving packets). This will effectively be a generic version of what the host controller specs look like.